src/opossum_lib/spdx/convert_to_opossum.py

```diff
@@ -1,95 +1,77 @@
 
-from opossum_lib.attribution_generation import (
+from opossum_lib.opossum.opossum_file import (
+    ExternalAttributionSource,
+    Metadata,
+    OpossumInformation,
+    OpossumPackage,
+    OpossumPackageIdentifier,
+    Resource,
+    ResourceType,
+    SourceInfo,
+)
+from opossum_lib.spdx.attribution_generation import (
     create_document_attribution,
     create_file_attribution,
     create_package_attribution,
     create_snippet_attribution,
 )
-from opossum_lib.constants import (
-    COMPRESSION_LEVEL,
+from opossum_lib.spdx.constants import (
     SPDX_FILE_IDENTIFIER,
     SPDX_PACKAGE_IDENTIFIER,
     SPDX_SNIPPET_IDENTIFIER,
 )
-from opossum_lib.helper_methods import (
+from opossum_lib.spdx.graph_generation import generate_graph_from_spdx
+from opossum_lib.spdx.helper_methods import (
     _create_file_path_from_graph_path,
     _get_source_for_graph_traversal,
     _node_represents_a_spdx_element,
     _replace_node_ids_with_labels_and_add_resource_type,
     _weakly_connected_component_sub_graphs,
 )
-from opossum_lib.opossum_file import (
-    ExternalAttributionSource,
-    Metadata,
-    OpossumInformation,
-    OpossumPackage,
-    OpossumPackageIdentifier,
-    Resource,
-    ResourceType,
-    SourceInfo,
-)
+from opossum_lib.spdx.tree_generation import generate_tree_from_graph
 
 
-def write_dict_to_file(
-    opossum_information: OpossumInformation, file_path: Path
-) -> None:
-    with ZipFile(
-        file_path, "w", compression=ZIP_DEFLATED, compresslevel=COMPRESSION_LEVEL
-    ) as z:
-        z.writestr("input.json", json.dumps(to_dict(opossum_information), indent=4))
-
-
-def to_dict(
-    element: Resource
-    | Metadata
-    | OpossumPackage
-    | OpossumInformation
-    | SourceInfo
-    | ExternalAttributionSource
-    | str
-    | int
-    | bool
-    | dict[str, OpossumPackage]
-    | dict[str, list[str]]
-    | list[str]
-    | None,
-) -> dict | str | list[str] | bool | int | None:
-    if isinstance(element, Resource):
-        return element.to_dict()
-    if isinstance(
-        element,
-        Metadata
-        | OpossumPackage
-        | OpossumInformation
-        | SourceInfo
-        | ExternalAttributionSource,
-    ):
-        result = []
-        for f in fields(element):
-            value = to_dict(getattr(element, f.name))
-            result.append((f.name, value))
-        return {k: v for (k, v) in result if v is not None}
-    elif isinstance(element, dict):
-        return {k: to_dict(v) for k, v in element.items()}
-    else:
-        return element
+def convert_spdx_to_opossum_information(filename: str) -> OpossumInformation:
+    try:
+        document: SpdxDocument = parse_file(filename)
+
+    except SPDXParsingError as err:
+        log_string = "\n".join(
+            ["There have been issues while parsing the provided document:"]
+            + [message for message in err.get_messages()]
+        )
+        logging.error(log_string)
+        sys.exit(1)
```

Comment:
Something we should think about in the future:
When we have multiple input files, do we want to abort as soon as a single conversion fails? In principle merges should be associative, so the user could decide to retry the conversion of the failed files later and then just merge the additional opossum files and get a result equivalent to what they would have gotten if all conversion+merge happened at once.

Related to #158